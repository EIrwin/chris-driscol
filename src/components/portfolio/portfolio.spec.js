// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Component from './portfolio.component';

let mockWorks = 0;
const createMockWork = () => {
  mockWorks++;
  return {
    title: `title ${mockWorks}`,
    imageUrl: `https://image/url/${mockWorks}`,
    location: `location ${mockWorks}`,
    duration: `duration ${mockWorks}`,
    description: `description ${mockWorks}`,
    technologies: `technologies ${mockWorks}`,
    subTitle: `subTitle ${mockWorks}`,
    link: `link ${mockWorks}`,
  };
};

const fixtures = {
  works: [
    createMockWork(),
    createMockWork(),
    createMockWork(),
    createMockWork(),
    createMockWork(),
  ],
};

it('renders correctly', () => {
  const tree = renderer.create(<Component {...fixtures} />).toJSON();
  expect(tree).toMatchSnapshot();
});
