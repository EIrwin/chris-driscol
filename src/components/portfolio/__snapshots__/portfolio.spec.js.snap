// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="home-section bg-light-gray portfolio"
  id="portfolio"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="text-center"
      >
        <h2
          className="section-heading"
        >
          My Work
        </h2>
        <p
          className="section-subheading text-muted"
        >
          These are just some of the things I have worked on over the years, some were done with the help of extremely talented colleagues.
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4 col-sm-6 portfolio-item"
      >
        <a
          className="portfolio-link"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          tabIndex={-1}
        >
          <div
            className="portfolio-hover"
          >
            <div
              className="portfolio-hover-content"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "height": "3em",
                    "verticalAlign": "middle",
                    "width": "3em",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m35.9 16.4v4.3q0 0.9-0.6 1.5t-1.5 0.7h-9.3v9.2q0 0.9-0.6 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.7-1.6v-9.2h-9.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h9.3v-9.3q0-0.9 0.7-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.6 1.5v9.3h9.3q0.9 0 1.5 0.6t0.6 1.5z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <img
            alt=""
            className="img-responsive"
            src="https://image/url/1"
          />
        </a>
        <div
          className="portfolio-caption"
        >
          <h4
            className="portfolio-title"
          >
            title 1
          </h4>
          <p
            className="text-muted"
          >
            location 1
          </p>
        </div>
      </div>
      <div
        className="col-md-4 col-sm-6 portfolio-item"
      >
        <a
          className="portfolio-link"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          tabIndex={-1}
        >
          <div
            className="portfolio-hover"
          >
            <div
              className="portfolio-hover-content"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "height": "3em",
                    "verticalAlign": "middle",
                    "width": "3em",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m35.9 16.4v4.3q0 0.9-0.6 1.5t-1.5 0.7h-9.3v9.2q0 0.9-0.6 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.7-1.6v-9.2h-9.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h9.3v-9.3q0-0.9 0.7-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.6 1.5v9.3h9.3q0.9 0 1.5 0.6t0.6 1.5z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <img
            alt=""
            className="img-responsive"
            src="https://image/url/2"
          />
        </a>
        <div
          className="portfolio-caption"
        >
          <h4
            className="portfolio-title"
          >
            title 2
          </h4>
          <p
            className="text-muted"
          >
            location 2
          </p>
        </div>
      </div>
      <div
        className="col-md-4 col-sm-6 portfolio-item"
      >
        <a
          className="portfolio-link"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          tabIndex={-1}
        >
          <div
            className="portfolio-hover"
          >
            <div
              className="portfolio-hover-content"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "height": "3em",
                    "verticalAlign": "middle",
                    "width": "3em",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m35.9 16.4v4.3q0 0.9-0.6 1.5t-1.5 0.7h-9.3v9.2q0 0.9-0.6 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.7-1.6v-9.2h-9.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h9.3v-9.3q0-0.9 0.7-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.6 1.5v9.3h9.3q0.9 0 1.5 0.6t0.6 1.5z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <img
            alt=""
            className="img-responsive"
            src="https://image/url/3"
          />
        </a>
        <div
          className="portfolio-caption"
        >
          <h4
            className="portfolio-title"
          >
            title 3
          </h4>
          <p
            className="text-muted"
          >
            location 3
          </p>
        </div>
      </div>
      <div
        className="clearfix hidden-sm"
      />
      <div
        className="col-md-4 col-sm-6 portfolio-item"
      >
        <a
          className="portfolio-link"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          tabIndex={-1}
        >
          <div
            className="portfolio-hover"
          >
            <div
              className="portfolio-hover-content"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "height": "3em",
                    "verticalAlign": "middle",
                    "width": "3em",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m35.9 16.4v4.3q0 0.9-0.6 1.5t-1.5 0.7h-9.3v9.2q0 0.9-0.6 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.7-1.6v-9.2h-9.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h9.3v-9.3q0-0.9 0.7-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.6 1.5v9.3h9.3q0.9 0 1.5 0.6t0.6 1.5z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <img
            alt=""
            className="img-responsive"
            src="https://image/url/4"
          />
        </a>
        <div
          className="portfolio-caption"
        >
          <h4
            className="portfolio-title"
          >
            title 4
          </h4>
          <p
            className="text-muted"
          >
            location 4
          </p>
        </div>
      </div>
      <div
        className="col-md-4 col-sm-6 portfolio-item"
      >
        <a
          className="portfolio-link"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          tabIndex={-1}
        >
          <div
            className="portfolio-hover"
          >
            <div
              className="portfolio-hover-content"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "height": "3em",
                    "verticalAlign": "middle",
                    "width": "3em",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m35.9 16.4v4.3q0 0.9-0.6 1.5t-1.5 0.7h-9.3v9.2q0 0.9-0.6 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.7-1.6v-9.2h-9.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h9.3v-9.3q0-0.9 0.7-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.6 1.5v9.3h9.3q0.9 0 1.5 0.6t0.6 1.5z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <img
            alt=""
            className="img-responsive"
            src="https://image/url/5"
          />
        </a>
        <div
          className="portfolio-caption"
        >
          <h4
            className="portfolio-title"
          >
            title 5
          </h4>
          <p
            className="text-muted"
          >
            location 5
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
